Eastern-Research-Group/csb-rebate-forms-app
# Feature/add bap query for 2023 submissions

## Related Issues:
* CSBAPP-252

## Main Changes:
* Fetches submission data from the BAP for 2023 submissions to support showing BAP submission statuses on the user's dashboard for 2023 submissions.
* Reduces a bit of duplicated code for combining submission data from the BAP and Formio that's now used for both 2022 and 2023 submissions.

## Steps To Test:
1. Create some FRF submissions for 2023
2. Request the BAP team to change the status of a few submissions
3. Ensure statuses are reflected as expected on the dashboard, along with all functionality related to statuses (e.g. "Edits Requested").
